aws/lambda/benchmark_regression_summary_report/lambda_function.py

```diff
@@ -0,0 +1,467 @@
+    def process(
+        self,
+        config_id: str,
+        end_time: dt.datetime,
+        cc: Optional[clickhouse_connect.driver.client.Client] = None,
+        args: Optional[argparse.Namespace] = None,
+    ):
+        def log_info(msg: str):
+            logger.info("[%s] %s", config_id, msg)
+
+        def log_error(msg: str):
+            logger.error("[%s] %s", config_id, msg)
+
+        # ensure each thread has its own clickhouse client. clickhouse client
+        # is not thread-safe.
+        if cc is None:
+            tlocal = threading.local()
+            if not hasattr(tlocal, "cc") or tlocal.cc is None:
+                if args:
+                    tlocal.cc = get_clickhouse_client(
+                        args.clickhouse_endpoint,
+                        args.clickhouse_username,
+                        args.clickhouse_password,
+                    )
+                else:
+                    tlocal.cc = get_clickhouse_client_environment()
+            cc = tlocal.cc
+        try:
+            config = get_benchmark_regression_config(config_id)
+            log_info(f"found config for config_id {config_id}")
+        except ValueError as e:
+            log_error(f"Skip process, Invalid config: {e}")
+            return
+        except Exception as e:
+            log_error(f"Unexpected error from get_benchmark_regression_config: {e}")
+            return
+
+        # check if the current time is > policy's time_delta + previous record_ts from summary_table
+        report_freq = config.policy.frequency
+        should_generate = self._should_generate_report(
+            cc, end_time, config_id, report_freq
+        )
+        if not should_generate:
+            log_info(
+                f"Skip generate report for time:{end_time} with frequency {report_freq.get_text()}, no data found",
+            )
+            return
+        else:
+            log_info(
+                f"Plan to generate report for time:{end_time} with frequency {report_freq.get_text()}..."
+            )
+        latest, ls, le = self.get_latest(config, end_time)
+        if not latest:
+            log_info(
+                f"no latest data found for time range [{ls},{le}] with frequency {report_freq.get_text()}..."
+            )
+            return
+
+        baseline, bs, be = self.get_baseline(config, end_time)
+        if not baseline:
+            log_info(
+                f"no baseline data found for time range [{bs},{be}] with frequency {report_freq.get_text()}..."
+            )
+            return
+
+    def get_latest(self, config: BenchmarkConfig, end_time: dt.datetime):
+        data_range = config.policy.range
+        latest_s = end_time - data_range.comparison_timedelta()
+        latest_e = end_time
+        latest_data = self._fetch_from_benchmark_ts_api(
+            config_id=config.id,
+            start_time=latest_s,
+            end_time=latest_e,
+            source=config.source,
+        )
+        logger.info(
```

Comment:
Nit: It's a bit weird to see `log_info` isn't used here, i.e. `log_info(f"found {leng()} # of data, with time range {time_range}")`